internal/test/random.go

```diff
@@ -0,0 +1,35 @@
+package test
+
+import (
+	"crypto/rand"
+	"math/big"
+)
+
+var (
+	strLen = 10
+	//nolint:gomnd // this avoids the golangci-lint "magic number" warning
+	n = big.NewInt(2)
+)
+
+// RandomString generates a cryptographically secure random string of a fixed
+// length. The string is composed of alphanumeric characters and is generated
+// using the crypto/rand library. The length of the string is determined by the
+// constant strLen.
+func RandomString() string {
+	var letters = []rune("abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ")
+	s := make([]rune, strLen)
+	max := big.NewInt(int64(len(letters)))
+	for i := range s {
+		randomInt, _ := rand.Int(rand.Reader, max)
+		s[i] = letters[randomInt.Int64()]
```

Comment:
I suggest we avoid `big` package and go with most basic implementation here e.g., https://stackoverflow.com/questions/22892120/how-to-generate-a-random-string-of-a-fixed-length-in-go

Reply:
Done